pkg/controller/upload-controller.go

```diff
@@ -114,12 +116,17 @@ func (r *UploadReconciler) Reconcile(req reconcile.Request) (reconcile.Result, e
 		log.V(1).Info("PVC has both clone and upload annotations")
 		return reconcile.Result{}, errors.New("PVC has both clone and upload annotations")
 	}
-
+	honorWaitForFirstConsumer, err := r.featureGates.HonorWaitForFirstConsumerEnabled()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
 	// force cleanup if PVC pending delete and pod running or the upload/clone annotation was removed
-	if (!isUpload && !isCloneTarget) || podSucceededFromPVC(pvc) || pvc.DeletionTimestamp != nil {
+	if !r.shouldReconcile(isUpload, isCloneTarget, honorWaitForFirstConsumer, pvc, log) || podSucceededFromPVC(pvc) || pvc.DeletionTimestamp != nil {
```

Comment:
Same suggestion here about passing in the featuregates instead of looking it up and passing the value.